dlt/common/destination/reference.py

```diff
@@ -493,19 +497,55 @@ def df(self, chunk_size: int = None) -> Optional[DataFrame]:
         """
         ...
 
-    def arrow(self, chunk_size: int = None) -> Optional[ArrowTable]: ...
+    # accessing data
+    def arrow(self, chunk_size: int = None) -> Optional[ArrowTable]:
+        """fetch arrow table of first 'chunk_size' items"""
+        ...
+
+    def ibis(self, chunk_size: int = None) -> Optional[IbisTable]: ...
+
+    def iter_df(self, chunk_size: int) -> Generator[DataFrame, None, None]:
+        """iterate over data frames tables of 'chunk_size' items"""
+        ...
+
+    def iter_arrow(self, chunk_size: int) -> Generator[ArrowTable, None, None]:
+        """iterate over arrow tables of 'chunk_size' items"""
+        ...
+
+    def iter_ibis(self, chunk_size: int) -> Generator[IbisTable, None, None]: ...
```

Comment:
how are you going to implement that? you'll need to create IbisTable on a chunk of data which is lazy evaluated. otherwise does not make sense

Reply:
removed it